Source/Evolve/WarpXPushType.H

```diff
@@ -0,0 +1,16 @@
+/* Copyright 2022 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_PUSHTYPE_H_
+#define WARPX_PUSHTYPE_H_
+
+enum struct PushType : int
+{
+    Explicit = 0,
+    Implicit
+};
```

Comment:
Could you add a few comments explaining what this controls:
```suggestion
enum struct PushType : int
// Controls the exact definition of the velocity v used in both the current deposition 
// and the update of the particle position
{
    Explicit = 0, // v^{n+1/2} = u^{n+1/2}/gamma^{n+1/2}
    Implicit // v^{n+1/2} = (u^{n+1} - u^{n})/(gamma^{n+1} - gamma^n)
    // The implicit definition is required in the current deposition in order to have 
    // exact energy conservation ; see Eq. 17 in Chen, JCP  407 (2020) 109228
};
```

Reply:
Hm, maybe we should also mention that it gathers the field at the position `(x^{n+1}+x^n)/2`.

Reply:
Your comments are good, though the change in how gamma is calculated is a side effect of the overall particle advance, with the explicit being a leap frog advance and the implicit being Crank-Nicholson. Perhaps this should be the comment.

Reply:
OK, sounds good. Feel free to adapt my suggestions accordingly.